connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -711,3 +728,261 @@ export function getDiscussionDocumentId(
+  isBatchSync: boolean,
+  loggerArgs: Record<string, string | number>
+) {
+  const localLogger = logger.child(loggerArgs);
+
+  const connector = await Connector.findOne({
+    where: {
+      connectionId: installationId,
+      dataSourceName: dataSourceConfig.dataSourceName,
+      workspaceId: dataSourceConfig.workspaceId,
+    },
+  });
+  if (!connector) {
+    throw new Error(`Connector not found (installationId: ${installationId})`);
+  }
+
+  const connectorState = await GithubConnectorState.findOne({
+    where: {
+      connectorId: connector.id,
+    },
+  });
+  if (!connectorState) {
+    throw new Error(`Connector state not found for connector ${connector.id}`);
+  }
+
+  if (!connectorState.codeSyncEnabled) {
+    localLogger.info("Code sync disabled for connector");
+    return;
+  }
+
+  const { tempDir, files, directories } = await processRepository({
+    installationId,
+    repoLogin,
+    repoName,
+    repoId,
+  });
+
+  try {
+    logger.info(
+      {
+        repoId,
+        filesCount: files.length,
+        directoriesCount: directories.length,
+        totalSize: files.reduce((acc, file) => acc + file.sizeBytes, 0),
+      },
+      "Downloaded Github repository for sync"
+    );
+
+    // From here everything happens locally or consists in upserting data. This is a big activity
+    // but if retried we're really just retrying downloading the repository externally and the
+    // upserts that succeeded before won't be retried as their associated GithubCodeFile object will
+    // have been updated. This means that while the syncing is not succeeedd we might have slightly
+    // incoherent state (files that moved will appear twice before final cleanup). This seems fine
+    // given that syncing stallness is already considered an incident.
+
+    const codeSyncStartedAt = new Date();
+    const rootInternalId = `github-code-${repoId}`;
+
+    const updatedDirectories: { [key: string]: boolean } = {};
+
+    for (const f of files) {
+      // Read file (files are 1MB at most).
+      const content = await fs.readFile(f.localFilePath);
+      const contentHash = blake3(content).toString("hex");
+
+      // Find file or create it with an empty contentHash.
+      let githubCodeFile = await GithubCodeFile.findOne({
+        where: {
+          connectorId: connector.id,
+          repoId,
+          documentId: f.documentId,
+        },
+      });
+
+      if (!githubCodeFile) {
+        githubCodeFile = await GithubCodeFile.create({
+          connectorId: connector.id,
+          repoId,
+          documentId: f.documentId,
+          parentInternalId: f.parentInternalId || rootInternalId,
+          fileName: f.fileName,
+          sourceUrl: f.sourceUrl,
+          contentHash: "",
+          createdAt: codeSyncStartedAt,
+          updatedAt: codeSyncStartedAt,
+          lastSeenAt: codeSyncStartedAt,
+        });
+      }
+
+      // We update if the file name or source url or content has changed.
```

Comment:
or parent it seems

Reply:
Yeah this is a bit of a lie since, if parents update then the documentId updates. But not adding the condition would feel weird. Will add a comment.

Reply:
Added checks for that. There was actually a small bug with the interplay with `rootInternalId`. See https://github.com/dust-tt/dust/pull/3089/commits/9f3b0fac03673b512ffa57a46cc509b550fac657